github/resource_github_repository_ruleset.go

```diff
@@ -0,0 +1,405 @@
+			},
+			"owner": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Owner of the repository. If not provided, the provider's default owner is used.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true, // TODO: Remove this when updating is implemented
+				Description: "The name of the Ruleset within the repository.",
+			},
+			"target": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true, // TODO: Remove this when updating is implemented
+				Description: "The target of the ruleset. Either branch or tag.",
+				ValidateFunc: validation.StringInSlice([]string{
+					"branch",
+					"tag",
+				}, false),
+			},
+			"enforcement": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true, // TODO: Remove this when updating is implemented
+				Description: "The enforcement level of the ruleset. One of active, disabled or evaluate. `evaluate` allows admins to test rules before enforcing them. Admins can view insights on the Rule Insights page (`evaluate` is only available with GitHub Enterprise).",
+				ValidateFunc: validation.StringInSlice([]string{
+					"active",
+					"disabled",
+					"evaluate",
+				}, false),
+			},
+			"conditions": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true, // TODO: Remove this when updating is implemented
+				MaxItems:    1,
+				Description: "Target branches/tags. Both an inclusion and exclusion list, supporting regexes as well as ALL branches/tags and the default branch",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// TODO: Should the default branch + ALL branches/tags have it's own field?
+						"include": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							ForceNew:    true, // TODO: Remove this when updating is implemented
+							Description: "Array of ref names or patterns to include. One of these patterns must match for the condition to pass. Also accepts `~DEFAULT_BRANCH` to include the default branch or `~ALL` to include all branches.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"exclude": {
+							Type:        schema.TypeSet,
+							Optional:    true,
+							ForceNew:    true, // TODO: Remove this when updating is implemented
+							Description: "Array of ref names or patterns to exclude. The condition will not pass if any of these patterns match.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+			"rule_creation": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true, // TODO: Remove this when updating is implemented
+				Description: "Only allow users with bypass permission to create matching refs.",
+			},
+			// TODO: Borken currently. When underlying sdk gets a version bump, its fixed
```

Comment:
`go-github` needs a version bump for this one I think. It's a weird one though, because both the [API docs](https://docs.github.com/en/enterprise-cloud@latest/rest/repos/rules?apiVersion=2022-11-28#get-a-repository-ruleset) and [`go-github`](https://github.com/google/go-github/blob/c030d43bc8e3003715a3de91972b1a594039d262/github/repos_rules.go#L70-L72) expects `update_allows_fetch_and_merge`, but when I ping the API, I get back a response looking like this (where the `update` field is the most important, the rest is just there for context):
```json
"rules": [
  {
    "type": "deletion"
  },
  {
    "type": "non_fast_forward"
  },
  {
    "type": "required_status_checks",
    "parameters": {
      "required_status_checks": [
        {
          "context": "SonarQube Code Analysis",
          "integration_id": 95502
        }
      ],
      "strict_required_status_checks_policy": true
    }
  },
  {
    "type": "pull_request",
    "parameters": {
      "require_code_owner_review": true,
      "require_last_push_approval": true,
      "dismiss_stale_reviews_on_push": true,
      "required_approving_review_count": 3,
      "required_review_thread_resolution": true
    }
  },
  {
    "type": "required_signatures"
  },
  {
    "type": "required_deployments",
    "parameters": {
      "required_deployment_environments": [
        "terraform"
      ]
    }
  },
  {
    "type": "required_linear_history"
  },
  {
    "type": "update"
  },
  {
    "type": "creation"
  }
],
```

I don't know why this is, or if I'm missing something

Reply:
Relevant discussion thread here: https://github.com/google/go-github/pull/2836